vnext/Shared/Networking/WinRTHttpResource.cpp

```diff
@@ -471,15 +471,25 @@ WinRTHttpResource::PerformSendRequest(HttpMethod &&method, Uri &&rtUri, IInspect
       auto inputStream = co_await response.Content().ReadAsInputStreamAsync();
       auto reader = DataReader{inputStream};
 
-      // #9510 - 10mb limit on fetch
-      co_await reader.LoadAsync(10 * 1024 * 1024);
+      // #9510 - We currently accumulate all incoming request data in 10MB chunks
+      uint32_t segmentSize = 10 * 1024 * 1024;
+      uint32_t length;
 
       // Let response handler take over, if set
       if (auto responseHandler = self->m_responseHandler.lock()) {
         if (responseHandler->Supports(reqArgs->ResponseType)) {
-          auto bytes = vector<uint8_t>(reader.UnconsumedBufferLength());
-          reader.ReadBytes(bytes);
-          auto blob = responseHandler->ToResponseData(std::move(bytes));
+          // #9510
+          vector<uint8_t> responseData{};
+          do {
+            co_await reader.LoadAsync(segmentSize);
+            length = reader.UnconsumedBufferLength();
+            auto data = vector<uint8_t>(length);
+            reader.ReadBytes(data);
+
+            responseData.insert(responseData.cend(), data.cbegin(), data.cend());
+          } while (length > 0);
```

Comment:
>} while (length > 0); [](http://example.com/codeflow?start=10&length=21)

Nit: Logic seems strange, in case of length ==0 we're going through the motions of allocating a 0-length vector and, presumably, reading 0 bytes and inserting a zero-length vector into the response data.
Would this work/be cleaner?

while(true) {
    co_await reader.LoadAsync(segmentSize);
    length = reader.UnconsumedBufferLength();
    if (length == 0) break;
    auto data = vector<uint8_t>(length);
    reader.ReadBytes(data);
    responseData.insert(responseData.cend(), data.cbegin(), data.cend());

}

Reply:
Turns out `reader.LoadAsync` co-returns the number of available bytes per segment.
This way, we can do:
```cpp
while (auto loaded = co_await reader.LoadAsync(segmentSize))
...
```